endpoint/endpoint.go

```diff
@@ -107,18 +109,25 @@ func (t Targets) IsLess(o Targets) bool {
 	return false
 }
 
+// ProviderSpecific holds configuration which is specific to individual DNS providers
+type ProviderSpecific map[string]string
```

Comment:
It's worth considering whether this should be a list of key-value pairs. Maps are supposed to be avoided per the API conventions: https://github.com/kubernetes/community/blob/master/contributors/devel/api-conventions.md#lists-of-named-subobjects-preferred-over-maps

Reply:
I think list of k-v pairs might be better because this is a contract many different types will have to implement.

Reply:
This field is added by the ongoing pr https://github.com/kubernetes-incubator/external-dns/pull/650. Its optional for the feature in this pr to work.
I would post the same comments in that pr. In this pr i have just added json tag for it.